docs/Preferences.md

```diff
@@ -0,0 +1,146 @@
+## File Associations
```

Comment:
I think the first preference which must be explained is the `xml.java.home` which is the more important. The idea is to add https://github.com/redhat-developer/vscode-xml#requirements but perhaps with more comment ?

I will add Server VM Arguments after that.

Reply:
I did the reordering you suggested. Please let me know if what I wrote for the cache settings makes sense.

Reply:
Very good explanation, thanks